src/main/java/com/academy/ui/components/SignInComponent.java

```diff
@@ -0,0 +1,112 @@
+package com.academy.ui.components;
+
+import com.academy.ui.pages.HomePage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+import org.openqa.selenium.support.PageFactory;
+
+public class SignInComponent extends BaseComponent {
+
+    @FindBy(how = How.ID, using = "email")
+    protected WebElement emailInput;
+
+    @FindBy(how = How.ID, using = "password")
+    protected WebElement passwordInput;
+
+    @FindBy(how = How.CSS, using = "button.greenStyle")
+    protected WebElement signInButton;
+
+
+    public SignInComponent(WebDriver driver, WebElement rootElement) {
+        super(driver, rootElement);
+        PageFactory.initElements(driver, this);
+    }
+
+    //emailInput
+    private void clickEmailInput() {
+        emailInput.click();
+    }
+
+    private void clearEmailInput() {
+        emailInput.clear();
+    }
+
+    private void setEmailInput(String email) {
+        emailInput.sendKeys(email);
+    }
+
+    //passwordInput
+    private void clickPasswordInput() {
+        passwordInput.click();
+    }
+
+    private void clearPasswordInput() {
+        passwordInput.clear();
+    }
+
+    private void setPasswordInput(String password) {
+        passwordInput.sendKeys(password);
+    }
+
+    //signInButton
+    private void clickSignInButton() {
+        signInButton.click();
+    }
+
+    public boolean isSignButtonActive() {
+        return signInButton.isEnabled();
+    }
+
+
+    // Functional
+
+    private void enterEmailInput(String email) {
+        clickEmailInput();
+        clearEmailInput();
+        setEmailInput(email);
+    }
```

Comment:
private SignInComponent enterEmailInput(String email) {
        clickEmailInput();
        clearEmailInput();
        setEmailInput(email);
        return this;
    }

Reply:
done